### PR TITLE
ONNXToTosa: Fix integer division for != 32 bits

### PR DESCRIPTION
**Pull request by mgehre-amd**
We don't need to be strict in the lowering as the TOSA dialect was extended to allow all integer types.

### CODE REVIEW

File: src/Conversion/ONNXToTOSA/DialectBuilder.cpp [hunk 1]
```diff
@@ -206,9 +206,8 @@ Value TosaBuilder::mul(mlir::Value &lhs, mlir::Value &rhs, int32_t shift) {
 Value TosaBuilder::intdiv(mlir::Value &lhs, mlir::Value &rhs) {
   Type lhsElementType = lhs.getType().cast<ShapedType>().getElementType();
   Type rhsElementType = rhs.getType().cast<ShapedType>().getElementType();
-  assert((lhsElementType.isSignlessInteger(32) &&
-             rhsElementType.isSignlessInteger(32)) &&
-         "Tosa DivOp needs 32-bit signless integer inputs");
```
Comment (cferry-AMD) [inline src/Conversion/ONNXToTOSA/DialectBuilder.cpp:211]: Just for the context, do you know why there was this restriction to 32-bit integers in the first place?

File: test/mlir/conversion/onnx_to_tosa/Math/Elementwise.mlir [hunk 2]
```diff
@@ -170,6 +170,16 @@ func.func @test_div(%arg0: tensor<13x21x1xi32>, %arg1: tensor<13x21x1xi32>) -> t
 
 // -----
 
+func.func @test_div_unsigned(%arg0: tensor<13x21x1xui8>, %arg1: tensor<13x21x1xui8>) -> tensor<13x21x1xui8> {
+  %0 = "onnx.Div"(%arg0, %arg1) : (tensor<13x21x1xui8>, tensor<13x21x1xui8>) -> tensor<13x21x1xui8>
+  "func.return"(%0) : (tensor<13x21x1xui8>) -> ()
+// CHECK-LABEL:  func @test_div_unsigned
+// CHECK-SAME:   ([[PARAM_0_:%.+]]: tensor<13x21x1xui8>, [[PARAM_1_:%.+]]: tensor<13x21x1xui8>) -> tensor<13x21x1xui8> {
+// CHECK-NEXT:      [[VAR_0_:%.+]] = tosa.div [[PARAM_0_]], [[PARAM_1_]] : (tensor<13x21x1xui8>, tensor<13x21x1xui8>) -> tensor<13x21x1xui8>
```
Comment (cferry-AMD) [inline test/mlir/conversion/onnx_to_tosa/Math/Elementwise.mlir:178]:
```suggestion
// CHECK-NEXT:      [[VAR_0_:%.+]] = tosa.int_div [[PARAM_0_]], [[PARAM_1_]] : (tensor<13x21x1xui8>, tensor<13x21x1xui8>) -> tensor<13x21x1xui8>
```

I didn't find `tosa.div` in [this spec](https://mlir.llvm.org/docs/Dialects/TOSA/#tosaint_div-mlirtosaintdivop)... maybe I'm out of date?

**Review by cferry-AMD — APPROVED** [review]

File: test/mlir/conversion/onnx_to_tosa/Math/Elementwise.mlir (hunk 2 above)
Comment (mgehre-amd) [reply test/mlir/conversion/onnx_to_tosa/Math/Elementwise.mlir:178]: No, this fork is out of date - the op was renamed to int_div on MLIR recently, but this branch still uses an older LLVM version.

File: src/Conversion/ONNXToTOSA/DialectBuilder.cpp (hunk 1 above)
Comment (mgehre-amd) [reply src/Conversion/ONNXToTOSA/DialectBuilder.cpp:211]: The TOSA spec only allows 32 bit integers, but MLIR allows more.

**Review by cferry-AMD — APPROVED** [review]
LGTM, thanks for the heads up on the out-of-date op mnemonic :)

**Review by TinaAMD — APPROVED** [review]
LGTM

**Pull request merged**